graphql-server/src/schemas/queries/samples.ts

```diff
@@ -492,3 +500,73 @@ export async function querySelectSampleDataForCacheUpdate(
     await session.close();
   }
 }
+
+/**
+ * Extracts the instrument model from a JSON string containing libraries data
+ * Returns the most recent instrument model based on run dates (confirmed with PMs)
+ */
+function getInstrumentModelByLibraries(
+  libraries: string | null
+): string | null {
+  if (libraries == null) return null;
+  try {
+    const libs = JSON.parse(libraries) as Array<{
+      runs?: Array<{
+        runMode?: string;
+        runDate?: string;
+      }>;
+    }>;
+    const latestRun = {
+      runMode: null as string | null,
+      runDate: null as Date | null,
+    };
+    for (const currLib of libs) {
+      if (!Array.isArray(currLib.runs) || currLib.runs.length === 0) continue;
+      for (const currRun of currLib.runs) {
+        if (!currRun.runMode || !currRun.runDate) continue;
+        const currRunDate = new Date(currRun.runDate);
+        if (!latestRun.runDate || currRunDate > latestRun.runDate) {
+          latestRun.runDate = currRunDate;
+          latestRun.runMode = currRun.runMode;
+        }
+      }
+    }
+    return latestRun.runMode;
```

Comment:
This block gets the most recent instrument model based on run dates.

In terms of implementation, I thought about combining JS's standard array methods like `flatMap`, `filter`, `reduce` here. Instead, I found this for-loop approach more efficient as we only iterate over each item once. Wanted to avoid adding any more time on this samples query than necessary.